src/middlewared/middlewared/etc_files/sysctl.py

```diff
@@ -0,0 +1,26 @@
+from pathlib import Path
+
+CONF_FILE = Path('/etc/sysctl.d/tunables.conf')
+
+
+def setup(middleware):
+    to_write = {}
+    for tun in middleware.call_sync('tunable.query', [['type', '=', 'SYSCTL']]):
+        if tun['enabled']:
+            # tunable is enabled and the variable's value set by the end-user doesn't match what the system's
+            # default value is so change it
+            to_write[tun['var']] = tun['value']
+        elif tun['orig_value']:
+            # tunable is disabled and we have an original value before it was created so update
+            to_write[tun['var']] = tun['orig_value']
+
+    if not to_write:
+        # no tunables or all disabled so make sure the config file is removed
+        CONF_FILE.unlink(missing_ok=True)
+    else:
+        with CONF_FILE.open('w') as f:
+            f.writelines('\n'.join(f'{tunable}={value}' for tunable, value in to_write.items()))
```

Comment:
Can values have spaces in them or quotes and in that case does the current configuration handles it appropriately ?

Reply:
It is impossible to validate every single sysctl that the linux kernel provides and even more so to validate every combination that the user provides. I'm just passing whatever the user gives to us as is. The onus is on them to make sure it's the correct syntax.